pkg/option/config.go

```diff
@@ -3123,8 +3139,23 @@ func InitConfig(programName, configName string) func() {
 			} else {
 				// replace deprecated fields with new fields
 				ReplaceDeprecatedFields(m)
-				err := MergeConfig(m)
-				if err != nil {
+
+				var confArgs []string
+				var flags = viper.AllKeys()
+				for key, value := range m {
+					if val := fmt.Sprintf("%v", value); val != "" {
+						// check whether the flag exists
+						if IsKnownFlag(flags, key) {
+							confArgs = append(confArgs, "--"+key+"="+val)
+						}
+					}
```

Comment:
I'm fairly certain these lines are causing the GKE failures you're seeing with the etcd kvstore config tests. I'm not exactly sure what's wrong, but these lines highlighted may be changing how the flags are interpreted by viper. I would try writing unit tests to have quicker feedback, rather than relying on the e2e tests. It would also make sense to validate some "harder" cases like `kvstore-opt` which is a map that contains inline JSON, and is the argument relevant in the errors you see.

Reply:
The weird thing here is that this actually affects the config passed to the agent, instead of just validating it. I guess this is caused by `cmd.ParseFlags` call on line 3154, which not only validates the flags, but also sets the values of target variables.